Wangscape/utils.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+#include <type_traits>
+#include <stdexcept>
+
+namespace
+{
+
+template<typename I>
+I _pow(I base, I exp)
+{
+    I result = 1;
+    for (int i = 0; i < std::numeric_limits<I>::digits; i++)
+    {
+        if (exp & 1)
+            result *= base;
+        else if (exp == 0)
+            break;
+        exp >>= 1;
+        base *= base;
+    }
+    return result;
+}
+
+}
+
+template <typename I>
+typename std::enable_if<std::is_unsigned<I>::value, I>::type pow(I base, I exp)
```

Comment:
Ideally this should be `template <typename I, typename J> I ipow(I, J)`. GTest's `TYPED_TEST` only allows one type parameter, but this could be circumvented by using `std::pair<I, J>` as the test template parameter and unpacking it with `TypeParam::first_type`/`TypeParam::second_type`.